gunrock/app/bc/bc_app.cu

```diff
@@ -1,425 +1,313 @@
+ * \return     double      Return accumulated elapsed times for all runs
  */
-void bc(
-    float*     bc_scores,
-    const int  num_nodes,
-    const int  num_edges,
-    const int* row_offsets,
-    const int* col_indices,
-    const int  source)
+template <
+    typename VertexT = int,
+    typename SizeT   = int,
+    typename GValueT = float,
+    typename BCValueT = GValueT>
+float bc(
+    const SizeT        num_nodes,
+    const SizeT        num_edges,
+    const SizeT       *row_offsets,
+    const VertexT     *col_indices,
+    //const GValueT     *edge_values,
+    const int          num_runs,
+          VertexT     *sources,
+          BCValueT   **bc_values,
+          BCValueT   **sigmas,
+          VertexT    **labels)
 {
-    struct GRTypes data_t;            // primitive-specific data types
-    data_t.VTXID_TYPE = VTXID_INT;    // integer vertex identifier
-    data_t.SIZET_TYPE = SIZET_INT;    // integer graph size type
-    data_t.VALUE_TYPE = VALUE_FLOAT;  // float attributes type
-
-    struct GRSetup *config = InitSetup(1, NULL);  // primitive-specific configures
-    config -> source_vertex[0] = source;        // source vertex to start
-
-    struct GRGraph *grapho = (struct GRGraph*)malloc(sizeof(struct GRGraph));
-    struct GRGraph *graphi = (struct GRGraph*)malloc(sizeof(struct GRGraph));
-
-    graphi->num_nodes   = num_nodes;  // setting graph nodes
-    graphi->num_edges   = num_edges;  // setting graph edges
-    graphi->row_offsets = (void*)&row_offsets[0];  // setting row_offsets
-    graphi->col_indices = (void*)&col_indices[0];  // setting col_indices
-
-    gunrock_bc(grapho, graphi, config, data_t);
-    memcpy(bc_scores, (float*)grapho->node_value1, num_nodes * sizeof(float));
-
-    if (graphi) free(graphi);
-    if (grapho) free(grapho);
+    typedef typename gunrock::app::TestGraph<VertexT, SizeT, GValueT,
+        gunrock::graph::HAS_CSR>
+        GraphT;
+    typedef typename GraphT::CsrT CsrT;
+
+    // Setup parameters
+    gunrock::util::Parameters parameters("bc");
+    gunrock::graphio::UseParameters(parameters);
+    gunrock::app::bc::UseParameters(parameters);
+    gunrock::app::UseParameters_test(parameters);
+    parameters.Parse_CommandLine(0, NULL);
+    parameters.Set("graph-type", "by-pass");
+    parameters.Set("num-runs", num_runs);
+    std::vector<VertexT> srcs;
+    for (int i = 0; i < num_runs; i ++)
+        srcs.push_back(sources[i]);
+    parameters.Set("srcs", srcs);
+
+    bool quiet = parameters.Get<bool>("quiet");
+    GraphT graph;
+    // Assign pointers into gunrock graph format
+    graph.CsrT::Allocate(num_nodes, num_edges, gunrock::util::HOST);
+    graph.CsrT::row_offsets   .SetPointer(row_offsets, gunrock::util::HOST);
+    graph.CsrT::column_indices.SetPointer(col_indices, gunrock::util::HOST);
+    //graph.CsrT::edge_values   .SetPointer(edge_values, gunrock::util::HOST);
+    graph.FromCsr(graph.csr(), true, quiet);
```

Comment:
Needs the full parameter list to avoid default parameter confusion or template mismatch:
```
 graph.FromCsr(graph.csr(), target, 0, quiet, true);
```

where `    gunrock::util::Location target = gunrock::util::HOST;` please refer to pr_app.cu for these changes.

Reply:
I don't think we need this call here -- PR uses `COO`, but BC uses CSR, so we can remove this line